File: app/src/main/java/net/crunkhouse/shaitzov/PlayingCardView.java
package net.crunkhouse.shaitzov;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

/**
 * A view that draws a PlayingCard model to an image.
 */

public class PlayingCardView extends LinearLayout {

    private TextView valueView;
    private ImageView suitView;
    private View backFaceView;

    public PlayingCardView(Context context) {
        super(context);
        init(context);
    }

    public PlayingCardView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    private void init(Context context) {
        setOrientation(LinearLayout.VERTICAL);
        setBackgroundResource(R.drawable.card_background);
        int width = DensityUtils.dpToPixel(context, 92);
        int height = DensityUtils.dpToPixel(context, 128);
        setMinimumHeight(height);
        setMinimumWidth(width);
        View rootView = inflate(context, R.layout.playing_card_view, this);
        valueView = (TextView) rootView.findViewById(R.id.value);
        suitView = (ImageView) rootView.findViewById(R.id.suit);
        backFaceView = rootView.findViewById(R.id.back_face);
    }

    public void setFaceDown(boolean faceDown) {
        if (faceDown) {
            valueView.setVisibility(View.GONE);
            suitView.setVisibility(View.GONE);
            backFaceView.setVisibility(VISIBLE);
        }
    }

    public void setCard(PlayingCard card) {
        valueView.setText(card.getValueName());
        Drawable suitImage;
        switch (card.getSuit()) {
            case CLUB:
                suitImage = getResources().getDrawable(R.drawable.suit_club);
                break;
            case DIAMOND:
                suitImage = getResources().getDrawable(R.drawable.suit_diamond);
                break;
            case SPADE:
                suitImage = getResources().getDrawable(R.drawable.suit_spade);
                break;
            default:
            case HEART:
                suitImage = getResources().getDrawable(R.drawable.suit_heart);
                break;
        }
        suitView.setImageDrawable(suitImage);
        suitView.setContentDescription(card.getSuit().toString());
    }
}

File: app/src/main/java/net/crunkhouse/shaitzov/ui/PlayingCardView.kt
package net.crunkhouse.shaitzov.ui

import android.content.Context
import android.graphics.drawable.Drawable
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout
import kotlinx.android.synthetic.main.playing_card_view.view.*

import net.crunkhouse.shaitzov.R
import net.crunkhouse.shaitzov.cards.PlayingCard
import net.crunkhouse.shaitzov.dp

/**
 * A view that draws a PlayingCard model to an image.
 */

class PlayingCardView : LinearLayout {

    constructor(context: Context) : super(context) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context)
    }

    private fun init(context: Context) {
        orientation = LinearLayout.VERTICAL
        setBackgroundResource(R.drawable.card_outline)
        val width = 60.dp(context)
        val height = 84.dp(context)
        minimumHeight = height
        minimumWidth = width
        inflate(context, R.layout.playing_card_view, this)
    }

    fun setFaceDown(faceDown: Boolean) {
        if (faceDown) {
            value.visibility = View.GONE
            suit.visibility = View.GONE
            back_face.visibility = View.VISIBLE
        }
    }

    fun setCard(card: PlayingCard) {
        value.text = card.valueName
        val suitImage: Drawable = when (card.suit) {
            PlayingCard.Suit.CLUB -> resources.getDrawable(R.drawable.suit_club)
            PlayingCard.Suit.DIAMOND -> resources.getDrawable(R.drawable.suit_diamond)
            PlayingCard.Suit.SPADE -> resources.getDrawable(R.drawable.suit_spade)
            PlayingCard.Suit.HEART -> resources.getDrawable(R.drawable.suit_heart)
            else -> resources.getDrawable(R.drawable.suit_heart)
        }
        suit.setImageDrawable(suitImage)
        suit.contentDescription = card.suit.toString()
    }
}
